tests/unit/src/test/scala/tests/codeactions/ExtractValueLspSuite.scala

```diff
@@ -164,12 +164,12 @@ class ExtractValueLspSuite
        |  method2({ 1 + <<2>> })
        |}
        |""".stripMargin,
-    ExtractValueCodeAction.title("{ 1 + 2 }"),
+    ExtractValueCodeAction.title("{ 1 + 2 }", "method"),
     """|object Main {
        |  def method2(i: Int) = ???
        |  val (newValue, newValue1) = (0, 1)
        |  def newValue0 = ???
-       |  val newValue2 = { 1 + 2 }
+       |  def newValue2 = { 1 + 2 }
```

Comment:
Do we need to extract those blocks as a function? val newValue2 = { 1 + 2 } should aldo be valid

Reply:
We can check if there is a `(` between method and `{ ... }` if that's what you mean

Reply:
Thanks! but I meant to, we don't need to extract those values as function, not only in this case but all the cases.

> When extracting from block apply, like Future { ...@@ } we have to extract as method, not as a value

I don't think so. I think we can extract things as values, do you have an example where we need to extract code as a function, instead of values?

You tested against `Future { method(5) }` to `def newValue = { method(5) }`, but it can be `val newValue = method(5)`.
Even if the code trying to extract has multiple statements like

```scala
Future {
  println("foo")
  "foo"
}
```

we can extract them as value like `val newValue = { println("foo"); "foo" }`

Reply:
I understand extracting as value may change the semantics (for example, if it's `IO`, extracting as value vs function change the semantics). However, most of the block applies (syntax) don't care if its function or value (for example, `Option { ... }`.

I think we don't necessarily need to preserve semantics by this command, it should be the users' responsibility.

Reply:
Yes, in most cases it doesn't matter if it's a value or function, but for `Future` don't we lose concurrency while extracting it as a value? How about a lazy val instead? 
Although you might be right that preserving semantics should be users responsibility, because there is (I think) no easy way  for us to distinguish where block apply extraction can change semantics

Reply:
>  preserving semantics should be users responsibility, because there is (I think) no easy way for us to distinguish where block apply extraction can change semantics

Yeah, I think there's no way to preserve the semantics (we can treat `Future` or `IO` as a special case, but it's a slippery slope. Though I don't come up with another case off the top of my head, there could be other cases where this refactoring breaks the semantics, we can't cover everything).

So my opinion is that we shouldn't support such a smart refactoring feature, and this PR would fix the title of code action.